### PR TITLE
CANS-186. Increased test coverage to 65% 

### PR DESCRIPTION
**Pull request by denys-davydov**
N.B. @jonbwalker is working on some other classes to cover with tests

### CODE REVIEW

File: client/src/components/Assessment/Domain.test.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  assessmentUnderSix={true}
+  i18n={clone(i18nPrototype)}
+  i18nAll={{'a': 'b'}}
+  onRatingUpdate={() => {}}
+  onConfidentialityUpdate={() => {}}
+/>;
+
+describe('<Domain />', () => {
+  it('renders', () => {
+    expect(() => shallow(domainComponentPrototype)).not.toThrow();
```
Comment (edgibbs) [inline client/src/components/Assessment/Domain.test.js]: This render has multiple states, with different behavior.  This just addresses rendering without blowing up.

File: client/src/components/Assessment/I18nHelper.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { clone } from 'lodash';
+import { getI18nByCode, getI18nValuesByPrefix } from './I18nHelper';
+
+describe('<I18nHelper />', () => {
+  describe('#getI18nByCode()', () => {
+    it('filters records by code and cuts the code off', () => {
```
Comment (edgibbs) [inline client/src/components/Assessment/I18nHelper.test.js]: Is code supposed to be local_code like en_US and that just isn't obvious from the test example?

File: client/src/components/Assessment/Item.test.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      />);
+      wrapper.setProps(clone(propsPrototype));
+      wrapper.instance().handleConfidentialityChange({target: {}})
+      expect(onConfidentialityUpdateMock.mock.calls.length).toBe(1);
+    });
+  });
+
+});
+
+const mountItem = item => {
```
Comment (edgibbs) [inline client/src/components/Assessment/Item.test.js]: This feels like it's floating at the bottom here, maybe within the describe where it's used or at least at the top of the test.

File: client/src/components/Assessment/Item.test.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import { clone } from 'lodash';
```
Comment (edgibbs) [inline client/src/components/Assessment/Item.test.js]: Prefer Object.assign for cloning over loading lodash.

File: client/src/components/Assessment/Item.test.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      const item = clone(itemPrototype);
+      item.rating_type = "BOOLEAN";
+      const wrapper = mountItem(item);
+      wrapper.setProps(clone(propsPrototype));
+      const findSelect = wrapper.find(Select);
+      expect(findSelect.children().get(0).props.children.length).toBe(4);
+
+      wrapper.find(Icon).simulate('click');
+      const expandedText = wrapper.text();
+      expect(expandedText).toMatch(/No = rating 0 descriptionYes = rating 1 description/);
```
Comment (edgibbs) [inline client/src/components/Assessment/Item.test.js]: This regex is hard to follow as to what it's proving, do we actually display a 0 and a 1?

File: client/src/components/Assessment/Item.test.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import { clone } from 'lodash';
+import { mount } from 'enzyme';
+import Item from './Item';
+import Icon from '@material-ui/core/Icon';
+import Radio from '@material-ui/core/Radio';
+import Select from '@material-ui/core/Select';
+
+const itemPrototype = {
```
Comment (edgibbs) [inline client/src/components/Assessment/Item.test.js]: Prototype is a pretty js code specific name, maybe just item, i18n, itemComponent.

File: client/src/setupTests.js [hunk 7]
```diff
@@ -2,3 +2,20 @@ import { configure } from 'enzyme';
...
+Object.keys(document.defaultView).forEach((property) => {
+  if (typeof global[property] === 'undefined') {
+    exposedProperties.push(property);
+    global[property] = document.defaultView[property];
+  }
+});
+
+global.navigator = {
+  userAgent: 'node.js'
+};
```
Comment (edgibbs) [inline client/src/setupTests.js]: no newline

File: client/src/setupTests.js [hunk 8]
```diff
@@ -2,3 +2,20 @@ import { configure } from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
 
 configure({ adapter: new Adapter() });
+
+var jsdom = require('jsdom').jsdom;
```
Comment (edgibbs) [inline client/src/setupTests.js]: OK, so this might be OK, but it rings off some alarm bells with testing anti-patterns.  I'm not sure why we're mocking globals like window, navigator, and document, seems like a YAGNI case.  We're also using 'var' instead of let or const.  Overall I'ld like to avoid mocking basic javascript components or at least mocking them much closer to where they're used.

File: client/src/components/Assessment/I18nHelper.test.js (hunk 2 above)
Comment (denys-davydov) [reply client/src/components/Assessment/I18nHelper.test.js]: This is a code of a domain or an item inside of the assessment

File: client/src/components/Assessment/Domain.test.js (hunk 1 above)
Comment (denys-davydov) [reply client/src/components/Assessment/Domain.test.js]: Yes, it is.

File: client/src/components/Assessment/Item.test.js (hunk 3 above)
Comment (denys-davydov) [reply client/src/components/Assessment/Item.test.js]: Moved it the file top where all the consts are initialized

File: client/src/components/Assessment/Item.test.js (hunk 4 above)
Comment (denys-davydov) [reply client/src/components/Assessment/Item.test.js]: Using {...object} approach now

File: client/src/components/Assessment/Item.test.js (hunk 6 above)
Comment (denys-davydov) [reply client/src/components/Assessment/Item.test.js]: Renamed the variables

File: client/src/components/Assessment/Item.test.js (hunk 5 above)
Comment (denys-davydov) [reply client/src/components/Assessment/Item.test.js]: "No = rating 0 description" is a description for No value
"Yes = rating 1 description" is a description for Yes value

File: client/src/setupTests.js (hunk 8 above)
Comment (denys-davydov) [reply client/src/setupTests.js]: The tests used to fail without mocking window and document, so I've used the approach as a workaround. I've tried it now, and it works without the workaround now, so removing the mocks.

File: client/src/components/Assessment/I18nHelper.test.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    it('filters records by prefix and returns values only', () => {
+      const inputI18n = {
+        'item10._title_': 'Title 10',
+        'item10._description_': 'Description 10',
+        'item11._title_': 'Title 11',
+        'item11._description_': 'Description 11',
+      };
+      const actualResult = getI18nValuesByPrefix(inputI18n, 'item10.');
+      const expectedResult = ['Title 10', 'Description 10'];
+      expect(actualResult).toEqual(expectedResult)
```
Comment (jonbwalker) [inline client/src/components/Assessment/I18nHelper.test.js]: because of the filtering logic in getI18nValuesByPrefix, we should probably test a failure for this one too to make sure there's not a false positive

File: client/src/components/Assessment/Item.test.js (hunk 5 above)
Comment (edgibbs) [reply client/src/components/Assessment/Item.test.js]: Weird but OK
